deploy/cdk_exec_policy/cdkExecPolicy.yaml

```diff
@@ -0,0 +1,180 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: Custom least privilege IAM policy for linking environments to dataall
+Parameters:
+  AwsAccountId:
+    Description: AWS AccountId of the dataall INFRASTRUCTURE account that we wish to link this environment with.
+    Type: String
+  PolicyName:
+    Description: IAM policy name (copy/paste from the UI)
+    Type: String
+    Default: 'DataAllCustomCDKPolicy'
+  EnvironmentResourcePrefix:
+    Description: The resource prefix value of the dataall environment. It MUST match the resource prefix that we use when we create the environment.
+    Type: String
+Resources:
+  CDKCustomExecutionPolicy0:
+    Type: 'AWS::IAM::ManagedPolicy'
+    Properties:
+      ManagedPolicyName: !Ref PolicyName
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Sid: KMS
+            Action:
+              - 'kms:CreateKey'
+            Effect: Allow
+            Resource: '*'
+          - Sid: ec2
+            Action:
+              - 'ec2:DescribeSecurityGroups'
+              - 'ec2:CreateSecurityGroup'
+              - 'ec2:createTags'
+              - 'ec2:DeleteSecurityGroup'
+              - 'ec2:RevokeSecurityGroupEgress'
+              - 'ec2:AuthorizeSecurityGroupIngress'
+              - 'ec2:AuthorizeSecurityGroupEgress'
+            Effect: Allow
+            Resource: '*'
+          - Sid: Athena
+            Effect: Allow
+            Action: 'athena:CreateWorkGroup'
+            Resource:
+              - !Sub 'arn:aws:athena:${AWS::Region}:${AWS::AccountId}:workgroup/*'
+          - Sid: IAM
+            Action:
+              - 'iam:CreatePolicy'
+              - 'iam:GetPolicy'
+            Effect: Allow
+            Resource:
+              - !Sub 'arn:aws:iam::${AWS::AccountId}:policy/*'
+          - Sid: IAMRole
+            Action:
+              - 'iam:AttachRolePolicy'
+              - 'iam:CreateRole'
+              - 'iam:CreateServiceLinkedRole'
+              - 'iam:GetRole'
+              - 'iam:GetRolePolicy'
+              - 'iam:PutRolePolicy'
+            Effect: Allow
+            Resource: '*'
+          - Sid: KMSCreateAlias
+            Action:
+              - 'kms:CreateAlias'
+            Effect: Allow
+            Resource:
+              - !Sub 'arn:aws:kms:${AWS::Region}:${AWS::AccountId}:alias/*'
+          - Sid: KMSKey
+            Action:
+              - 's3:PutBucketAcl'
+              - 's3:PutBucketNotification'
+            Effect: Allow
+            Resource:
+              - !Sub 'arn:aws:s3:::${EnvironmentResourcePrefix}-logging-*'
+          - Sid: ReadBuckets
+            Action:
+              - 'kms:CreateAlias'
+              - 'kms:CreateGrant'
+              - 'kms:Decrypt'
```

Comment:
Why do we need Encrypt/Decrypt permissions for this role. It is a role to create resources right?